core/src/main/java/org/apache/calcite/rel/type/RelDataTypeFactoryImpl.java

```diff
@@ -562,6 +562,37 @@ public RelDataType createDecimalQuotient(
     return null;
   }
 
+  public RelDataType decimalOf(RelDataType type) {
+    Preconditions.checkArgument(SqlTypeUtil.isNumeric(type) || SqlTypeUtil.isNull(type));
+    SqlTypeName typeName = type.getSqlTypeName();
+    assert typeName != null;
+    switch (typeName) {
+    case DECIMAL:
+      return type;
+    case TINYINT:
+      return createSqlType(SqlTypeName.DECIMAL, 3, 0);
```

Comment:
Have you forget the null attribute of value `type`, function `createSqlType` will always return a type that is never null

Reply:
Yes, i do not sync the nullability attribute here, it is better and more safe to add and sync nullability.